grafast/grafserv/src/middleware/graphql.ts

```diff
@@ -452,22 +452,34 @@ export const makeGraphQLHandler = (
       // Validate that the body is of the right shape
       body = validateGraphQLBody(parsedBody);
     } catch (e) {
-      if (e instanceof SafeError) {
-        throw e;
-      } else if (
-        typeof e.statusCode === "number" &&
-        e.statusCode >= 400 &&
-        e.statusCode < 600
-      ) {
-        throw e;
-      } else {
-        // ENHANCE: should maybe handle more specific issues here. See examples:
-        // https://graphql.github.io/graphql-over-http/draft/#sec-application-json.Examples
-        throw httpError(
-          400,
-          `Parsing failed, please check that the data you're sending to the server is correct`,
-        );
-      }
+      return {
+        type: "graphql",
+        request,
+        dynamicOptions,
+        statusCode:
+          e instanceof SafeError
+            ? e.extensions.statusCode
+            : typeof e.statusCode === "number" &&
+              e.statusCode >= 400 &&
+              e.statusCode < 600
+            ? e.statusCode
+            : 400,
+        contentType: chosenContentType,
+        payload: maskPayload({
+          errors: [
+            new GraphQLError(
+              e.message ??
+                `Parsing failed, please check that the data you're sending to the server is correct`,
+              null,
+              undefined,
+              undefined,
+              undefined,
+              e,
+              undefined,
+            ),
+          ],
+        }),
+      };
```

Comment:
Digging a little deeper, this is currently handled by `handleGraphQLHandlerError` in `grafast/grafserv/src/core/base.ts` which is called as part of `_processRequest`. But that isn't called by the fastify handler... Feels like it should be self-contained.

Suggest you:
1. revert this
2. move `handleGraphQLHandlerError` into this file instead (and remove the associated logic from base.ts)
3. Rename `makeGraphQLHandler` to `_makeGraphQLHandlerInternal` and remove the `export`
4. Add a new function `makeGraphQLHandler` and export it; it should just call `_makeGraphQLHandlerInternal` and add the `.catch(...)` expression that you deleted from `base.ts`

Reply:
New commit